src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterReceiver.cs

```diff
@@ -22,62 +22,61 @@ internal class EventHubAdapterReceiver : IQueueAdapterReceiver, IQueueCache
     {
         private static readonly TimeSpan ReceiveTimeout = TimeSpan.FromSeconds(5);
 
-        private readonly ICacheDataAdapter<EventData, CachedEventHubMessage> dataAdapter;
         private readonly EventHubPartitionConfig config;
+        private readonly IObjectPool<FixedSizeBuffer> bufferPool;
 
-        private PooledQueueCache<EventData, CachedEventHubMessage> cache;
+        private PooledQueueCache<EventData, CachedEventHubMessage, string> cache;
         private EventHubReceiver receiver;
+        private EventHubPartitionCheckpoint checkpoint;
 
-        private class Cursor : IQueueCacheCursor
-        {
-            private readonly PooledQueueCache<EventData, CachedEventHubMessage> cache;
-            private readonly object cursor;
-            private IBatchContainer current;
+        public int MaxAddCount { get { return 1000; } }
 
-            public Cursor(PooledQueueCache<EventData, CachedEventHubMessage> cache, Guid streamGuid, string streamNamespace, StreamSequenceToken token)
-            {
-                this.cache = cache;
-                cursor = cache.GetCursor(streamGuid, streamNamespace, token);
-            }
+        public EventHubAdapterReceiver(EventHubPartitionConfig partitionConfig, IObjectPool<FixedSizeBuffer> bufferPool, Logger log)
+        {
+            config = partitionConfig;
+            this.bufferPool = bufferPool;
+        }
 
-            public void Dispose()
-            {
-            }
+        public async Task Initialize(TimeSpan timeout)
+        {
+            var dataAdapter = new EventHubDataAdapter(bufferPool);
+            cache = new PooledQueueCache<EventData, CachedEventHubMessage, string>(dataAdapter);
+            dataAdapter.PurgeAction = cache.Purge;
+            checkpoint = new EventHubPartitionCheckpoint(config.CheckpointSettings, config.StreamProviderName, config.Partition);
+            await checkpoint.InitTableAsync();
+            string offset = await checkpoint.Load();
+            receiver = await CreateReceiver(config, offset);
+        }
 
-            public IBatchContainer GetCurrent(out Exception exception)
+        public async Task<IList<IBatchContainer>> GetQueueMessagesAsync(int maxCount)
+        {
+            if (receiver == null)
             {
-                exception = null;
-                return current;
+                await Task.Delay(1); // prevent freespin
```

Comment:
Good catch.

There was a time when receivers would sometimes continue to be called even after they were shutdown.  This delay was added to the original receiver as a mitigation to that issue.  I pulled over the behavior without consideration of why it was there, but the issue was resolved, so it's no longer needed.